src/lib/output/themes/default/templates/hierarchy.tsx

```diff
@@ -1,46 +1,17 @@
 import type { DefaultThemeRenderContext } from "../DefaultThemeRenderContext";
-import type { PageEvent } from "../../../events";
-import { JSX } from "../../../../utils";
-import { getHierarchyRoots } from "../../lib";
-import type { DeclarationReflection, ProjectReflection } from "../../../../models";
 
-function fullHierarchy(
-    context: DefaultThemeRenderContext,
-    root: DeclarationReflection,
-    seen = new Set<DeclarationReflection>(),
-) {
-    if (seen.has(root)) return;
-    seen.add(root);
+import { JSX } from "../../../../utils";
 
-    // Note: We don't use root.anchor for the anchor, because those are built on a per page basis.
-    // And classes/interfaces get their own page, so all the anchors will be empty anyways.
-    // Full name should be safe here, since this list only includes classes/interfaces.
+export function hierarchyTemplate(context: DefaultThemeRenderContext) {
     return (
-        <li>
-            <a id={root.getFullName()} class="tsd-anchor"></a>
-            <a href={context.urlTo(root)}>
-                {context.icons[root.kind]()}
-                {root.name}
-            </a>
-            <ul>
-                {root.implementedBy?.map((child) => {
-                    return child.reflection && fullHierarchy(context, child.reflection as DeclarationReflection, seen);
-                })}
-                {root.extendedBy?.map((child) => {
-                    return child.reflection && fullHierarchy(context, child.reflection as DeclarationReflection, seen);
-                })}
-            </ul>
-        </li>
-    );
-}
+        <section class="tsd-panel tsd-hierarchy" id="tsd-hierarchy-container" data-base={context.relativeURL("./")}>
+            <h4>{context.i18n.theme_hierarchy_summary()}</h4>
 
-export function hierarchyTemplate(context: DefaultThemeRenderContext, props: PageEvent<ProjectReflection>) {
-    return (
-        <>
-            <h2>{context.i18n.theme_class_hierarchy_title()}</h2>
-            {getHierarchyRoots(props.project).map((root) => (
-                <ul class="tsd-full-hierarchy">{fullHierarchy(context, root)}</ul>
-            ))}
-        </>
+            <ul class="tsd-hierarchy tsd-full-hierarchy">
+                <li class="tsd-hierarchy-item">
+                    <span>{context.i18n.theme_loading()}</span>
```

Comment:
This page should not need JS, including the full hierarchy in a place where users without it enabled can see it is useful.

Reply:
I was under the assumption that TypeDoc always requires JavaScript, as the entire navigation section is non-functional without it.

Assuming that was the case, this change was made to improve size on disk (otherwise the same data would be in both `hierarchy.html` and `assets/hierarchy.js`) and to make the hierarchy trees match. Currently interface hierarchies with multiple roots get partially rendered multiple times, where as the new rendering logic renders the entire tree once.

It would be possible to duplicate the rendering logic and make the `hierarchy.html` page static while fixing multi-root hierarchies, but that would still unnecessarily increase the file size if JavaScript is still required..

Reply:
While the user experience is significantly degraded, it's still possible to navigate through the docs, the one link that exists in the navigation takes you to a page which lists all of its children, so it can be browsed via that method.

It will make the generated docs slightly larger, but including that information twice is acceptable, 500x times, less so :)

Reply:
Changed